From d05dd3f20fed08159875b9d4a78ab27ec02cfab6 Mon Sep 17 00:00:00 2001
From: Zdenek Kabelac <zkabelac@redhat.com>
Date: Wed, 9 May 2012 12:12:21 +0000
Subject: [PATCH 2/3] Fix regression in for_each_sub_lv

pool_lv is not a sub lv in terms for this function.
It has caused problem with renaming thin_volume, where it has tried to
rename pool LV as well.

(cherry picked from commit 98f2e3d974bcceaf6dbac9f80f42a0eaf6512059)

Conflicts:
	WHATS_NEW
---
 WHATS_NEW               |    4 ++++
 lib/metadata/lv_manip.c |    7 -------
 2 files changed, 4 insertions(+), 7 deletions(-)

--- a/WHATS_NEW
+++ b/WHATS_NEW
@@ -1,3 +1,7 @@
+Version 2.02.95-wrs
+================================
+  Fix lvrename for thin volumes (regression in for_each_sub_lv() 2.02.89).
+
 Version 2.02.95 - 6th March 2012
 ================================
   If unspecified, adjust thin pool metadata and chunk size to fit into 128MB.
--- a/lib/metadata/lv_manip.c
+++ b/lib/metadata/lv_manip.c
@@ -2814,13 +2814,6 @@ int for_each_sub_lv(struct cmd_context *
 				return_0;
 		}
 
-		if (seg->pool_lv) {
-			if (!fn(cmd, seg->pool_lv, data))
-				return_0;
-			if (!for_each_sub_lv(cmd, seg->pool_lv, fn, data))
-				return_0;
-		}
-
 		if (seg->metadata_lv) {
 			if (!fn(cmd, seg->metadata_lv, data))
 				return_0;
